### ISSUE TITLE
Add rav1e to package managers

### ISSUE THREAD

**Issue by EwoutH** [xiph/rav1e#886]
Not directly related to the codebase, but it would be very useful to have rav1e available in the most used package managers. These for example:
 - [ ] [APT](https://wiki.debian.org/Apt) (Debian / Ubuntu)
 - [ ] [Homebrew](https://brew.sh/index_nl) (macOS)
 - [ ] [Chocolatey](https://chocolatey.org/) (Windows)

**Comment by EwoutH** [comment]
@lu-zero Homebrew [requires](https://docs.brew.sh/Formula-Cookbook) at least one tagged version. Is there a release that can be tagged as 0.0.1 or something?

**Comment by lu-zero** [comment]
I'm fine with the weekly tag approach, but I'd like to have everybody ack it.

**Comment by lu-zero** [comment]
(once we have a tag we could also produce crav1e builds/packages)

**Comment by lu-zero** [comment]
Here a [prerelease](https://github.com/xiph/rav1e/releases/tag/20190129) so we can see what is missing packaging-wise.

**Comment by petterreinholdtsen** [comment]
I created https://bugs.debian.org/920842 asking for a Debian package.  Tagged releases will make it more likely. :)

**Comment by jbeich** [comment]
Are you interested only in select few repositories or any? For example:
[![Packaging status](https://repology.org/badge/vertical-allrepos/rav1e.svg)](https://repology.org/metapackage/rav1e/versions)

**Comment by EwoutH** [comment]
@jbeich The more the merrier :).

For CI the Ubuntu LTS versions (14.04, 16.04 and 18.04) and Debian Stable are the most important.

For general use Manjaro, Mint, MX Linux, elementary and Fedora would also be nice.

**Comment by cuviper** [comment]
cc @ignatenkobrain for Fedora too.

**Comment by vi** [comment]
Maybe at least publish pre-built executables for various linuxes to Github releases?

**Comment by cromefire** [comment]
> Maybe at least publish pre-built executables for various linuxes to Github releases?

Having one for the latest Ubuntu LTS would be nice as it's pretty common

**Comment by vi** [comment]
It it can be just a fully static musl build, like this:

https://vi-server.org/pub/rav1e_af9d4ab

    SHA256 (rav1e_af9d4ab) = 73c6f959ee38324608648a3e09b9cf6e04b02a2c359ddea84e20dc22f1e464fd

**Comment by jbeich** [comment]
> https://vi-server.org/pub/rav1e_af9d4ab

Works fine even on FreeBSD via [linuxulator](https://www.freebsd.org/doc/en/books/handbook/linuxemu.html). Tested in a clean jail without any Linux userland bits installed.

**Comment by EwoutH** [comment]
@jbeich does the packaging status badge get updated automatically?

**Comment by jbeich** [comment]
@EwoutH, yes, Repology [badges](https://repology.org/metapackage/rav1e/badges) are dynamic. Not sure how frequent repository scans are: probably, every few hours. Try checking out badges for packages e.g., [cbindgen](https://repology.org/badge/vertical-allrepos/cbindgen.svg).

**Comment by petterreinholdtsen** [comment]
Btw, is it possible to build rav1e from the git repo on Debian without
downloading extra code from the Internet?  Software builds downloading
random code from the Internet rub me the wrong way in addition to
failing the 'desert island' test for free software, from
<URL: https://people.debian.org/~bap/dfsg-faq.html >:

  The Desert Island test.
  
  Imagine a castaway on a desert island with a solar-powered
  computer. This would make it impossible to fulfill any requirement to
  make changes publicly available or to send patches to some particular
  place. This holds even if such requirements are only upon request, as
  the castaway might be able to receive messages but be unable to send
  them. To be free, software must be modifiable by this unfortunate
  castaway, who must also be able to legally share modifications with
  friends on the island.

In practice it require the complete source to be available from Debian.

-- 
Happy hacking
Petter Reinholdtsen

**Comment by petterreinholdtsen** [comment]
[Ximin Luo]
> Debian Rust packaging is already done with a cargo wrapper tool that
> avoids network access, see
> https://salsa.debian.org/rust-team/debcargo-conf/ - rav1e can submit a
> PR for itself and any of its dependencies not already in Debian.

Aha.  Perhaps you can help make build rules for a rav1e Debian package?
Or at least update <URL: https://bugs.debian.org/920842 > with details
on how to do it?  I know nothing about rust myself.

-- 
Happy hacking
Petter Reinholdtsen

**Comment by cromefire** [comment]
As a first ting wouldn't it be good to have it on [crates.io](https://crates.io) (the offical rust package repo) so anyone can use `cargo install rav1e`? All listed package managers are restricted to a specific OS, while cargo is not.

**Comment by eclipseo** [comment]
I'm already taking care of aom and dav1d for Fedora, I'll take care of rav1e as well when there is a point release.

**Comment by vibhoothiiaanand** [comment]
@eclipseo Hi,

As we have a 0.1 release of rav1e, will it be possible for you to take care of rav1e too?

**Comment by barrbrain** [comment]
The following dependencies need to be packaged before rav1e for Debian:
```
dpkg-checkbuilddeps: error: Unmet build dependencies:
librust-arbitrary-0.2+default-dev
librust-arg-enum-proc-macro-0.3+default-dev
librust-arrayvec-0.5+default-dev
librust-better-panic-0.2+default-dev
librust-bitstream-io-0.8+default-dev
librust-err-derive-0.2+default-dev (>= 0.2.1-~~)
librust-interpolate-name-0.2+default-dev (>= 0.2.2-~~)
librust-ivf-0.1+default-dev
librust-nasm-rs-0.1+default-dev
librust-noop-proc-macro-0.1+default-dev
librust-paste-0.1+default-dev
librust-pretty-env-logger-0.3+default-dev
librust-scan-fmt-0.2+default-dev (>= 0.2.3-~~)
librust-signal-hook-0.1+default-dev (>= 0.1.9-~~)
librust-simd-helpers-0.1+default-dev
librust-vergen-3+default-dev
librust-y4m-0.4+default-dev
```
Line-breaks added for readability.

**Comment by barrbrain** [comment]
We are waiting on https://github.com/Homebrew/homebrew-core/pull/46975 so that we can include the dev artifacts in the formula. This will enable the bottle to be used for building against the C-ABI.

**Comment by cryptomilk** [comment]
I've packaged rav1e for openSUSE which is on the way to openSUSE Tumbleweed.

Fedora is next on my list.

**Comment by shmerl** [comment]
Are there any plans to package Debian dependencies? I tried using `avifenc` in Debian recently, but it has no decoder:

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=976349

**Comment by petterreinholdtsen** [comment]
[shmerl]
> Are there any plans to package Debian dependencies?

I suggest bringing this up with the Debian Multimedia team,
<URL: https://wiki.debian.org/DebianMultimedia >.

-- 
Happy hacking
Petter Reinholdtsen

**Comment by sebastinas** [comment]
I'm currently in the process of packaging rav1e for Debian. The current blockers for me are at least the dependency on `vergen` version 3 instead of version 6.

**Comment by lu-zero** [comment]
We are using a custom one, a patch to update to the latest is welcome though.

**Comment by sebastinas** [comment]
The custom vergen version does not seem to be included in the rav1e version from `crates.io`, though. In any case, the use of vergen seems in the package build seems to be counter productive anyway. Even if I locally patch it to use version 6, it then fails because we are not building from a git repository.

**Comment by lu-zero** [comment]
Also the current vergen is 7.

**Comment by lu-zero** [comment]
there is some overlap with `built`, maybe we could drop one.

**Comment by lu-zero** [comment]
We might, let see if the output matches.

**Comment by sebastinas** [comment]
rav1e is now available in Debian: https://tracker.debian.org/pkg/rust-rav1e
